### PR TITLE
Integrate all modules into pipeline

### PR DESCRIPTION
**Pull request by FWDekker**
The pipeline is here!

* Integrates all modules into the pipeline.
* Adds a beautiful CLI for Schaapi.
* Adds a cute little smoke test.

### CODE REVIEW

**Comment by gandreadis** [comment]
@FWDekker Any idea why the smoke test is failing on both CI services? (works on my machine :P)

**Comment by FWDekker** [comment]
@gandreadis Wait, it works on your machine? That's interesting, because I forgot to add the JAR that is used by the user project as a dependency, causing the compilation to fail (hence why Travis shows a `MavenInvocationException`). Interestingly it also works on my second machine. Pushing the missing JAR in 3... 2... 1

**Comment by FWDekker** [comment]
@gandreadis It was actually even less impressive than that: I forgot to install Maven before running it. Of course, on our machines it was already installed, so no problemo, but on CI? Nope.

Well, I _hope_ that was the problem. We'll see.

**Comment by gandreadis** [comment]
@FWDekker Hmm, now getting a `ZipException` in the CI builds...

File: src/main/kotlin/org/cafejojo/schaapi/Schaapi.kt [hunk 1]
```diff
@@ -1,18 +1,155 @@
...
+        cmd.getOptionOrDefault("pattern_detector_minimum_count", DEFAULT_PATTERN_DETECTOR_MINIMUM_COUNT).toInt()
+    ).findFrequentSequences()
+        .filter { IncompleteInitPatternFilter.retain(it) }
+
+    val classGenerator = SootClassGenerator(DEFAULT_PATTERN_CLASS_NAME)
+    patterns.forEachIndexed { index, pattern ->
+        classGenerator.generateMethod("pattern$index", pattern)
+    }
+
+    SootClassWriter.writeToFile(classGenerator.sootClass, outputPatterns.absolutePath)
```
Comment (casperboone) [inline src/main/kotlin/org/cafejojo/schaapi/Schaapi.kt]: Replace with `classGenerator.writeToFile(outputPatterns.absolutePath)`

File: src/main/kotlin/org/cafejojo/schaapi/Schaapi.kt [hunk 2]
```diff
@@ -1,18 +1,155 @@
...
+        fullyQualifiedClassName = DEFAULT_PATTERN_CLASS_NAME,
+        classpath = outputPatterns.absolutePath + File.pathSeparator + library.classpath,
+        outputDirectory = outputTests.absolutePath,
+        generationTimeoutSeconds = testGeneratorTimeout,
+        processStandardStream = if (testGeneratorEnableOutput) System.out else null,
+        processErrorStream = if (testGeneratorEnableOutput) System.out else null
+    ).run()
+}
+
+private fun buildOptions(): Options {
```
Comment (casperboone) [inline src/main/kotlin/org/cafejojo/schaapi/Schaapi.kt]:
```kotlin
private fun buildOptions() = Options()
    .addOption(....)
```

File: src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt [hunk 3]
```diff
@@ -2,24 +2,41 @@ package org.cafejojo.schaapi
 
 import org.assertj.core.api.Assertions.assertThat
 import org.jetbrains.spek.api.Spek
-import org.jetbrains.spek.api.dsl.describe
 import org.jetbrains.spek.api.dsl.it
+import java.io.File
+import java.nio.file.Files
 
+/**
+ * E2E-test for Schaapi.
```
Comment (casperboone) [inline src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt]: Can we please just write end to end here? 😇

File: src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt [hunk 4]
```diff
@@ -2,24 +2,41 @@ package org.cafejojo.schaapi
...
+ */
 internal class SchaapiTest : Spek({
```
Comment (casperboone) [inline src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt]: Maybe rename to SchaapiSmokeTest for clarity about the meaning of the test

File: src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt [hunk 5]
```diff
@@ -2,24 +2,41 @@ package org.cafejojo.schaapi
...
+            "--test_generator_timeout", "3"
+        ))
+
+        assertThat(target.resolve("patterns/RegressionTest.class")).isFile()
+        assertThat(target.resolve("tests/evosuite-tests/RegressionTest_ESTest.java")).isFile()
     }
 })
+
+private fun getResourcePath(path: String) =
+    SchaapiTest::class.java.getResource(path).path
```
Comment (casperboone) [inline src/test/kotlin/org/cafejojo/schaapi/SchaapiTest.kt]: Can be on previous line

File: subprojects/project-compiler/gradle.properties [hunk 6]
```diff
@@ -1,2 +1,2 @@
-mavenInvokerVersion  = 3.0.0
-zip4jVersion         = 1.3.2
+mavenInvokerVersion = 3.0.0
+ztZipVersion        = 1.13
```
Comment (casperboone) [inline subprojects/project-compiler/gradle.properties]: Can we specify the minor version as well?

Comment (FWDekker) [reply subprojects/project-compiler/gradle.properties]: zt-zip doesn't have a minor version

**Pull request merged**